### PR TITLE
UCT/UD: simple AI/MD congestion avoidance

### PR DESCRIPTION
**Pull request by alex-mikheev**
@yosefe @brminich please take a look

### CODE REVIEW

File: src/uct/ib/ud/base/ud_def.h [hunk 1]
```diff
@@ -16,11 +16,24 @@
...
 #define UCT_UD_QP_HASH_SIZE     256    
 #define UCT_UD_MAX_SGE          2
-#define UCT_UD_MAX_WINDOW       1024 /* TODO: make it config param */
 #define UCT_UD_TX_MODERATION    64
 #define UCT_UD_MIN_INLINE       48
 #define UCT_UD_HASH_SIZE        997
 
+/* congestion avoidance settings */
+/* adaptive increase/multiplicative decrease */
+#define UCT_UD_CA_AI_VALUE      1   /* windoe += AI_VALUE */
```
Comment (yosefe) [inline src/uct/ib/ud/base/ud_def.h]: typo in comment

Comment (shamisp) [inline src/uct/ib/ud/base/ud_def.h]: I would suggest to add a writeup about the protocol

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1533/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/927/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1538/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/932/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1552/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/945/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/947/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1554/ for details.

**Comment by yosefe** [comment]
:+1:

**Pull request merged**